### PR TITLE
Indexer views

### PR DESCRIPTION
**Pull request by blueNoteb5**
* Add a list and a detail view for the ``Indexer`` model
* The list view is searchable/filterable

refs #5, #6

### CODE REVIEW

File: django/cantusdb_project/main_app/tests/test_views.py [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+        self.assertEquals(response.status_code, 200)
+        # Check object_list (which has the whole queryset, not paginated) instead of
+        # indexers which is paginated and might not contain random_indexer if it
+        # is not on the first page
+        self.assertTrue(random_indexer in response.context["paginator"].object_list)
+
+        # Search for a three letter slice of the first name
+        first_name = random_indexer.first_name
+        first_name_length = len(first_name)
+        if first_name_length <= 3:
```
Comment (JRegimbal) [inline django/cantusdb_project/main_app/tests/test_views.py]: Why not just test `if first_name_length > 3:` and remove the pass?

File: django/cantusdb_project/requirements.txt [hunk 2]
```diff
@@ -1,6 +1,22 @@
+appdirs==1.4.4
 asgiref==3.2.7
-Django==3.0.7
+attrs==19.3.0
+black==19.10b0
+click==7.1.2
+Django==3.0.6
```
Comment (JRegimbal) [inline django/cantusdb_project/requirements.txt]: Was the downgrade intentional?

File: django/cantusdb_project/main_app/templates/indexer_list.html [hunk 3]
```diff
@@ -0,0 +1,45 @@
+{% extends "base.html" %}
+{% block content %}
+<div class="mr-3 p-3 col-md-10 mx-auto bg-white rounded">
+    <h3>List of Indexers</h3>
+    {% with count=indexers.count %}
+    <small>Displaying 1-{{count}} of <b>{{count}}</b></small>
```
Comment (JRegimbal) [inline django/cantusdb_project/main_app/templates/indexer_list.html]: Is there no pagination for this view?

**Review by JRegimbal — COMMENTED** [review]
Overall looks good with very useful comments! A few minor questions but that's it.

File: django/cantusdb_project/main_app/templates/indexer_list.html (hunk 3 above)
Comment (blueNoteb5) [reply django/cantusdb_project/main_app/templates/indexer_list.html]: There is, that's a mistake, I'll fix it

File: django/cantusdb_project/requirements.txt (hunk 2 above)
Comment (blueNoteb5) [reply django/cantusdb_project/requirements.txt]: Nope, GitHub had updated it but when I did ``pip freeze`` it went back. I'll fix it, good catch

File: django/cantusdb_project/main_app/tests/test_views.py (hunk 1 above)
Comment (blueNoteb5) [reply django/cantusdb_project/main_app/tests/test_views.py]: Good point, I'll rewrite it

**Pull request merged**